addons/mail/static/src/core/common/thread_model.js

```diff
@@ -785,7 +783,7 @@ export class Thread extends Record {
                     { name: newName }
                 );
             } else if (this.supportsCustomChannelName) {
-                this.custom_channel_name = newName;
+                this.selfMember.custom_channel_name = newName;
```

Comment:
Note: this whole method should be moved to discuss folder as it is only related to channels (can be a separate commit/PR).